classes/log/store.php

```diff
@@ -63,15 +64,59 @@ protected function is_event_ignored(event_base $event) {
         return $isdisabledevent;
     }
 
+    /**
+     * get eventid from logstore_standard_log
+     * @param event_base $event
+     * @return int
+     *
+     */
+    protected function get_event_id($event) {
+        global $DB;
+
+        $sqlparams = [];
+        $sqlparams['eventname'] = $event->eventname;
+        $sqlparams['component'] = $event->component;
+        $sqlparams['action'] = $event->action;
+        $sqlparams['target'] = $event->target;
+        $sqlparams['objecttable'] = $event->objecttable;
+        $sqlparams['objectid'] = $event->objectid;
+        $sqlparams['timecreated'] = $event->timecreated;
+        $sqlparams['userid'] = $event->userid;
+        $sqlparams['anonymous'] = $event->anonymous;
+
+        $rows = $DB->get_records_sql("SELECT MAX(ID) AS id
+                    FROM {logstore_standard_log}
+                    WHERE eventname = :eventname
+                    AND component = :component
+                    AND action = :action
+                    AND target = :target
+                    AND (objecttable = :objecttable OR objecttable IS NULL)
+                    AND (objectid = :objectid OR objectid IS NULL)
+                    AND timecreated = :timecreated
+                    AND userid = :userid
+                    AND anonymous = :anonymous", $sqlparams);
```

Comment:
Please use Moodle doc format styling when you create an SQL.

Reply:
What will happen if standard log is disabled?

Reply:
> What will happen if standard log is disabled?

Good point - I hadn't thought of that

Reply:
Since replaying historical logs at the moment relies on data being inserted from logstore_standard_log I think if they want to use this feature they need to have it enabled.